src/io/GeantSetup.hh

```diff
@@ -0,0 +1,90 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file GeantSetup.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "celeritas_config.h"
+#include "base/Assert.hh"
+
+// Geant4 forward declarations
+class G4VPhysicalVolume;
+class G4RunManager;
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct a Geant 4 run manager and populate internal Geant4 physics.
+ *
+ * This is usually passed directly into \c GeantImporter . It hides geant
+ * includes from the rest of the code.
+ */
+class GeantSetup
+{
+  public:
+    //! Selection of physics processes
+    enum class PhysicsList
+    {
+        em_basic,    //!< Celeritas demo loop
+        em_standard, //!< G4EmStandardPhysics
+        ftfp_bert,   //!< Full physics
+    };
+
+  public:
+    // Construct from a GDML file and physics options
+    GeantSetup(const std::string& gdml_filename, PhysicsList physics);
+
+    // Default constructor
+    GeantSetup() = default;
+
+    // Get the world detector volume
+    inline const G4VPhysicalVolume* world() const;
+
+    //! True if we own a run manager
+    explicit operator bool() const { return static_cast<bool>(run_manager_); }
+
+  private:
+    struct RMDeleter
+    {
+        void operator()(G4RunManager*) const;
+    };
+    using RMUniquePtr = std::unique_ptr<G4RunManager, RMDeleter>;
+
+    RMUniquePtr              run_manager_{nullptr};
+    const G4VPhysicalVolume* world_{nullptr};
```

Comment:
Would it be better to make world_ a smart ptr?

Reply:
No, Geant4 always has ownership of the world volume -- we can't make it "smart" unless Geant4 allows us to handle memory management.